activerecord/lib/active_record/enum.rb

```diff
@@ -225,6 +225,8 @@ def detect_enum_conflict!(enum_name, method_name, klass_method = false)
           raise_conflict_error(enum_name, method_name)
         elsif !klass_method && method_defined_within?(method_name, _enum_methods_module, Module)
           raise_conflict_error(enum_name, method_name, source: "another enum")
+        elsif klass_method && method_defined_within?(method_name, Relation)
+          raise_conflict_error(enum_name, method_name, source: ActiveRecord::Relation.name)
```

Comment:
The error raised here currently says:

> [...] this will generate a instance method "records", which is already defined by ActiveRecord::Relation.

We should pass `type: "class"`, so that it says:

> [...] this will generate a **class** method "records"

Reply:
What do you think of moving this code up by 4 lines, so that both of the `if klass_method` checks are next to each other? It might have been easier to spot the `type: "class"` inconsistency if that was the case.

Reply:
We're referencing `Relation` on the line above, and `ActiveRecord::Relation` here; we can use `Relation` in both places, since this code is inside the `ActiveRecord` module.

Reply:
Thanks man!

one question if I may,
When I call a model scope method, Don't I get a relation instance, and when calling a scope method on this relation, it's a class method, where does the delegation happen ?

Reply:
`Relation#method_missing` delegates methods to the model class if it responds to them:
https://github.com/rails/rails/blob/v5.1.4/activerecord/lib/active_record/relation/delegation.rb#L88-L90